src/peer_connection.cpp

```diff
@@ -1488,10 +1490,11 @@ namespace libtorrent {
 
 		if (is_disconnecting()) return;
 
+		int const block_size = t->block_size();
 		auto const dlq_iter = std::find_if(
 			m_download_queue.begin(), m_download_queue.end()
-			, std::bind(match_request, std::cref(r), std::bind(&pending_block::block, _1)
-			, t->block_size()));
+			, [&r, block_size](pending_block const& pb)
+			{ return match_request(r, pb.block, block_size); });
```

Comment:
is `match_request` used in other places too? if not, should it just be moved directly into the lambda instead?

Reply:
no, I will move it here then